Builds/Test.py

```diff
@@ -75,55 +95,69 @@
     '--keep_going', '-k',
     action='store_true',
     help='Keep going after one configuration has failed.',
- )
+)
 
 parser.add_argument(
     '--silent', '-s',
     action='store_true',
     help='Silence all messages except errors',
- )
+)
 
 parser.add_argument(
     '--verbose', '-v',
     action='store_true',
     help=('Report more information about which commands are executed and the '
           'results.'),
- )
+)
 
 parser.add_argument(
     '--test', '-t',
     default='',
     help='Add a prefix for unit tests',
- )
+)
+
+parser.add_argument(
+    '--nonpm', '-n',
+    action='store_true',
+    help='Do not run npm tests',
+)
+
+parser.add_argument(
+    '--clean', '-c',
+    action='store_true',
+    help='delete all build artifacts after testing',
+)
 
 parser.add_argument(
     'scons_args',
     default=(),
     nargs='*'
-    )
+)
 
 ARGS = parser.parse_args()
 
-def shell(*cmd, **kwds):
-    "Execute a shell command and return the output."
-    silent = kwds.pop('silent', ARGS.silent)
-    verbose = not silent and kwds.pop('verbose', ARGS.verbose)
+
+def shell(cmd, args=(), silent=False):
+    """"Execute a shell command and return the output."""
+    silent = ARGS.silent or silent
+    verbose = not silent and ARGS.verbose
     if verbose:
-        print('$', ' '.join(cmd))
-    kwds['shell'] = IS_WINDOWS
+        print('$' + cmd, *args)
+
+    command = (cmd,) + args
 
     process = subprocess.Popen(
-        cmd,
+        command,
         stdin=subprocess.PIPE,
         stdout=subprocess.PIPE,
         stderr=subprocess.STDOUT,
-        **kwds)
+        shell=IS_WINDOWS)
     lines = []
     count = 0
     for line in process.stdout:
-        lines.append(line)
+        lines.append(line.decode())
         if verbose:
-            print(line, end='')
+            print(line.decode(), end='')
```

Comment:
Random first google result: http://pythoncentral.io/encoding-and-decoding-strings-in-python-3-x/

On my machine (using python3) all returned lines are actually byte arrays (https://www.python.org/dev/peps/pep-3112/), meaning the output is not formatted at all and just consists of b'foo\n',b'bar\n',b'baz\n' and so on. Windows traditionally has... issues with encodings. Which Python version, Windows version and Python distribution are you using? I'd like to keep the decode() in, if possible, maybe again with an "IS_WINDOWS" switch in front that sets an encoding that your machine uses.